src/checker/reqwest_checker.rs

```diff
@@ -214,115 +224,125 @@ impl ReqwestChecker {
-                    tls_timing = full_duration
+        if all_bytes.len() > max_bytes {
+            tracing::info!("aborting huge body");
+            break all_bytes;
+        }
+    }
+}
+
+fn to_check_status_and_reason(
+    assert_cache: &assertions::cache::Cache,
+    response: Result<Response, reqwest::Error>,
+    check: &ScheduledCheck,
+    body_bytes: &[u8],
+    subscription_id: &Uuid,
+) -> (CheckStatus, Option<CheckStatusReason>) {
+    match response {
+        Ok(r) => match r.status().is_success() {
+            true => {
+                if let Some(assertion) = &check.get_config().assertion {
+                    let comp_assert = assert_cache.get_or_compile(assertion);
+
+                    match comp_assert {
+                        Ok(assertion) => {
+                            let result =
+                                assertion.eval(r.status().as_u16(), r.headers(), body_bytes);
+
+                            match result {
+                                Ok(result) => {
+                                    if result {
+                                        (CheckStatus::Success, None)
+                                    } else {
+                                        (
+                                            CheckStatus::Failure,
+                                            Some(CheckStatusReason {
+                                                status_type:
+                                                    CheckStatusReasonType::AssertionFailure,
+                                                description: "Assertion failed".to_string(),
+                                            }),
+                                        )
+                                    }
+                                }
+                                Err(err) => (
+                                    CheckStatus::Failure,
+                                    Some(CheckStatusReason {
+                                        status_type: CheckStatusReasonType::AssertionError,
+                                        description: err.to_string(),
+                                    }),
+                                ),
+                            }
+                        }
+                        Err(err) => {
+                            tracing::warn!(
+                                "a bad assertion made it to compile from {} : {}",
+                                subscription_id,
+                                err.to_string(),
+                            );
+                            (
+                                CheckStatus::Failure,
+                                Some(CheckStatusReason {
+                                    status_type: CheckStatusReasonType::AssertionError,
+                                    description: err.to_string(),
+                                }),
+                            )
+                        }
+                    }
```

Comment:
Would like to see if we could reduce the [pyramid of doom](https://en.wikipedia.org/wiki/Pyramid_of_doom_(programming)) effect here